RallyHereIntegration/Source/RallyHereIntegration/Private/RH_PlayerNotifications.cpp

```diff
@@ -41,12 +47,49 @@ void URH_PlayerNotifications::StartStreamingLatestNotifications(const FString& C
 	StreamingCursor = Cursor;
 	StreamingPlayerUuid = GetRHPlayerUuid();
 
+	// reset the delay on long polls
+	NextLongPollDelay = 0.f;
+
 	bIsStreaming = true;
 
 	if (PollingType == FRH_NotificationPollingTypes::LongPoll)
 	{
+		RestartLongPollLoop();
+	}
+}
+
+void URH_PlayerNotifications::RestartLongPollLoop(bool bPreviousPollSuccessful)
+{
+	if (bPreviousPollSuccessful)
+	{
+		// clear poll delay and restart immediately if previous poll was successful
+		NextLongPollDelay = 0.f;
 		StartLongPoll();
 	}
+	else
+	{
+		// increment delay and restart on a timer (2 second base increase, capped at max time)
+		NextLongPollDelay = FMath::Min(2.f + NextLongPollDelay * 2.f, MaxLongPollDelay);
+		if (PlayerInfo != nullptr)
+		{
+			auto GISS = PlayerInfo->GetPlayerInfoSubsystem()->GetGameInstanceSubsystem();
+			if (GISS != nullptr)
+			{
+				auto pWorld = GISS->GetWorld();
+
+				if (pWorld != nullptr)
+				{
+					pWorld->GetTimerManager().SetTimer(LongPollDeferralHandle, this, &URH_PlayerNotifications::StartLongPoll, NextLongPollDelay, false);
+				}
+			}
+		}
+
+		// if we failed to set a deferal timer on a failure, stop streaming
+		if (!LongPollDeferralHandle.IsValid())
```

Comment:
How possible is it for us to get here?  Curious if this could get hit if we are like transitioning levels and notification api is having issues or something.

Reply:
Should be very unlikely.  Right now this object is not necessarily directly owned by a player info, but its our only way back up the chain.  So its possible some use cases may not have it.